### DISCUSSION TITLE
Planning project architecture - set up local environment

### DISCUSSION THREAD

**Discussion by couchpotato78** [Pythagora-io/gpt-pilot#471] [Q&A]
Hi,

After answering all questions, the next fase is the 'planning project architecture',  GPT Pilot asks to please setup my environment so that the technologies listed can be utilized.

So I have a clean install of Ubuntu 22.04/3 LTS and many of the listed technologies are not installed.

Following the install and configure instruction a virtual environment is created, must I install all the technologies in this environment?
And looking at the list of technologies, is there a simple way to install this?
- Node.js
- Express
- Passport
 - jsonwebtoken
 - MySQL
 - Sequelize
 - React
 - Axios
 - Recharts
 - Socket.io
 - Bootstrap
 
So I think that Node.js and MySQL are system-wide installations.
For Python-related packages, you use a virtual environment (pilot-env), however, most of the above technologies listed are JavaScript-based and do not require a Python virtual environment.
I think these will be set up on a per-project basis, using Node.js and npm.

Are the following steps correct?

1. Install Node.js and MySQL system-wide.

2. Set up (or choose) my project directory
Is this the project directory that GPT-Pilot creates under the folder workspace?
For me: /GPT-Pilot/gpt-pilot/workspace/app/tests/
Is this the correct folder '/GPT-Pilot/gpt-pilot/workspace/app/' ?
Use npm init in my project folder to start a new Node.js project.
Install Node.js packages (express, passport, etc.) in my project folder.
Create a React app inside my project folder.

3. Database and ORM (Object Relational Mapping) Setup:
Configure MySQL on my system or while GPT-Pilot do this?
Use Sequelize as an ORM in your Node.js project to interact with MySQL or while GPT-Pilot do this?

I would like to create my first app using GPT-Pilot, hope you guys van help me, thank you.

**Comment by couchpotato78** [comment]
In the meantime i have tried to setup the required environment, GPT-Pilot asks when it's done.
Strange enough: GPT-Pilot ask me to approve the following command: npm install express sequelize passport jsonwebtoken axios socket.io` with 30000ms timeout

This means GPT pilot is asking me to install the needed technologies.

So it's not necessary to setup the required technologies?

**Comment by BruceRayWilson** [comment]
I am not on the team.

Yes, gpt-pilot will install the packages for you.
